engines/twine/animations.h

```diff
@@ -0,0 +1,157 @@
+	uint8 *animTable[NUM_ANIMS];
+	/** Table with all loaded animations sizes */
+	uint32 animSizeTable[NUM_ANIMS];
+
+	/** Rotation by anim and not by engine */
+	int16 processRotationByAnim; // processActorVar5
+	/** Last rotation angle */
+	int16 processLastRotationAngle; // processActorVar6
+	/** Current process actor index */
+	int16 currentlyProcessedActorIdx;
+
+	/** Current step X coornidate */
+	int16 currentStepX;
+	/** Current step Y coornidate */
+	int16 currentStepY;
+	/** Current step Z coornidate */
+	int16 currentStepZ;
+	/** Current actor anim extra pointer */
+	uint8 *currentActorAnimExtraPtr;
+
+	/** Pointer to current animation keyframe */
+	uint8 *keyFramePtr;
+	/** Pointer to last animation keyframe */
+	uint8 *lastKeyFramePtr;
+
+	uint8 *animBuffer1;
+	uint8 *animBuffer2;
+
+	/** Set animation keyframe
+	@param keyframIdx Animation keyframe index
+	@param animPtr Pointer to animation
+	@param bodyPtr Body model poitner
+	@param animTimerDataPtr Animation time data */
+	int32 setAnimAtKeyframe(int32 keyframeIdx, uint8 *animPtr, uint8 *bodyPtr, AnimTimerDataStruct *animTimerDataPtr);
+
+	/** Get total number of keyframes in animation
+	@param animPtr Pointer to animation */
+	int32 getNumKeyframes(uint8 *animPtr);
+
+	/** Get first keyframes in animation
+	@param animPtr Pointer to animation */
+	int32 getStartKeyframe(uint8 *animPtr);
+
+	/** Set new body animation
+	@param animIdx Animation index
+	@param animPtr Animation pointer
+	@param bodyPtr Body model poitner
+	@param animTimerDataPtr Animation time data */
+	int32 setModelAnimation(int32 animIdx, uint8 *animPtr, uint8 *bodyPtr, AnimTimerDataStruct *animTimerDataPtr);
+
+	/** Get entity anim index (This is taken from File3D entities)
+	@param animIdx Entity animation index
+	@param actorIdx Actor index */
+	int32 getBodyAnimIndex(int32 animIdx, int32 actorIdx);
+
+	/** Stock animation - copy the next keyFrame from a different buffer
+	@param animPtr Animation pointer
+	@param bodyPtr Body model poitner
+	@param animTimerDataPtr Animation time data */
+	int32 stockAnimation(uint8 *animPtr, uint8 *bodyPtr, AnimTimerDataStruct *animTimerDataPtr);
+
+	/** Verify animation at keyframe
+	@param animIdx Animation index
+	@param animPtr Animation pointer
+	@param bodyPtr Body model poitner
+	@param animTimerDataPtr Animation time data */
+	int32 verifyAnimAtKeyframe(int32 animPos, uint8 *animPtr, uint8 *bodyPtr, AnimTimerDataStruct *animTimerDataPtr);
+
+	/** Initialize animation
+	@param newAnim animation to init
+	@param animType animation type
+	@param animExtra animation actions extra data
+	@param actorIdx actor index */
+	int32 initAnim(int32 newAnim, int16 animType, uint8 animExtra, int32 actorIdx);
+
+	/** Process acotr animation actions
+	@param actorIdx Actor index */
+	void processAnimActions(int32 actorIdx);
+
+	/** Process main loop actor animations
+	@param actorIdx Actor index */
+	void processActorAnimations(int32 actorIdx);
+};
+
+}
```

Comment:
You're already using // namespace TwinE comments at the end of the namespaces elsewhere, so it's a good practice to have the comment for all of them

Reply:
woops - they were added by the formatter - not sure why it is missing here. will re-run the formatter